pkg/controllers/cloud/configmap_controller.go

```diff
@@ -59,37 +59,66 @@ func (r *ConfigMapReconciler) Reconcile(_ context.Context, req ctrl.Request) (ct
 		r.Log.Error(err, "exiting nephe-controller")
 		os.Exit(1)
 	}
+
+	if err := r.setCloudSyncInterval(newControllerConfig.CloudSyncInterval); err != nil {
+		// Unable to continue as interval is not valid.
+		r.Log.Error(err, "exiting nephe-controller")
+		os.Exit(1)
+	}
+
 	return ctrl.Result{}, nil
 }
 
 // setCloudResourcePrefix sets the CloudResourcePrefix, returns error when invalid.
 func (r *ConfigMapReconciler) setCloudResourcePrefix(prefix string) error {
 	if len(prefix) == 0 {
-		r.Log.Info("Setting CloudResourcePrefix to default", "Old", r.ControllerConfig.CloudResourcePrefix,
-			"New", config.DefaultCloudResourcePrefix)
+		SetLastSyncTime()
```

Comment:
this should be set only when the interval is changed.
It should be called in setCloudSyncInterval().

Reply:
Solved.